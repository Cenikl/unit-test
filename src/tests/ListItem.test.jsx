import { render, fireEvent } from "@testing-library/react";
import renderer from 'react-test-renderer';
import { ListItem } from "../ListItem";

const mockOnCheck = jest.fn();

describe('ListItem', () => {
    it('display value correctly', () => {
        const { getByText } = render(
            <ListItem
                id='list-item-test'
                checkable={true}
                onCheck={mockOnCheck}
                item='Lorem ipsum dolor sit amet consectetur'
            />
        );

        const value = getByText('Lorem ipsum dolor sit amet consectetur');
        expect(value).toBeInTheDocument();
    });
    
    it('checkbox is shown', () => {
        const { getByTestId } = render(
            <ListItem
                id='list-item-1'
                checkable={true}
                onCheck={mockOnCheck}
                item='Lorem ipsum dolor sit amet consectetur'
            />
        );
        const node = getByTestId('test-list-item-1-container');
        const value = getByTestId('test-list-item-1');
        expect(value).toBeInTheDocument();
        expect(node.children).toHaveLength(2);
    });

    it('checkbox is hidden', () => {
        const { getByTestId, debug } = render(
            <ListItem
                id='list-item-1'
                checkable={false}
                onCheck={mockOnCheck}
                item='Lorem ipsum dolor sit amet consectetur'
            />
        );
        const node = getByTestId('test-list-item-1-container');
        expect(node.children).toHaveLength(1);
    });
    
    //TODO: implement this
    it('callback is called', () => {
        const { getByTestId } = render(
            <ListItem
                id='list-item-1'
                checkable={true}
                onCheck={mockOnCheck}
                item='Lorem ipsum dolor sit amet consectetur'
            />
        );
        const checking = getByTestId('test-list-item-1');
        fireEvent.click(checking);
        expect(mockOnCheck).toBeCalledTimes(1);
    });

    //TODO: implement this
    it('callback is not called when not checkable', () => {
        const { getByTestId } = render(
            <ListItem
                id='list-item-1'
                checkable={false}
                onCheck={mockOnCheck}
                item='Lorem ipsum dolor sit amet consectetur'
            />
        );
        const node = getByTestId('test-list-item-1-container');
        
        expect(node.children).toHaveLength(1);
        expect(mockOnCheck).toBeCalledTimes(0);
    });

    //TODO: implement this
    it('matches saved snapshot', () => {
        const tree = renderer.create(<ListItem id='list-item-test'checkable={true}onCheck={mockOnCheck}item='Lorem ipsum dolor sit amet consectetur'/>
).toJSON();
        expect(tree).toMatchSnapshot();
    });

    
    it('matches saved snapshot 2', () => {
        const tree = renderer.create(
            <ListItem
                id='list-item-test'
                checkable={false}
                onCheck={mockOnCheck}
                item='Lorem ipsum dolor sit amet consectetur'
            />
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });
});