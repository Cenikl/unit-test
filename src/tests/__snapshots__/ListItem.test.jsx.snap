// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListItem matches saved snapshot 1`] = `
<li
  className="list-item"
  data-testid="test-list-item-test-container"
>
  <input
    data-testid="test-list-item-test"
    id="list-item-test"
    onChange={[Function]}
    type="checkbox"
  />
  <label
    htmlFor="list-item-test"
  >
    Lorem ipsum dolor sit amet consectetur
  </label>
</li>
`;

exports[`ListItem matches saved snapshot 2 1`] = `
<li
  className="list-item"
  data-testid="test-list-item-test-container"
>
  <label
    htmlFor="list-item-test"
  >
    Lorem ipsum dolor sit amet consectetur
  </label>
</li>
`;
